### ISSUE TITLE
Search-R1 Performance Regression in verl v0.7.1 and Reproducibility Issues in v0.8.0

### ISSUE THREAD

**Issue by zhenlongDai** [verl-project/verl#7394]
## Description

I have encountered some reproducibility and performance regression issues when migrating Search-R1 experiments across different versions of `verl`.

As `verl` evolves, scripts and configurations that worked with earlier versions can no longer be directly executed with newer versions due to changes in APIs, configuration structures, and implementations. While some manual adaptation is expected, I also observe a significant performance gap even when trying to keep the training setup and hyperparameters as consistent as possible.

In particular:

- The Search-R1 implementation works well with the earlier `verl` version (e.g., v0.5.0).
- When reproducing the same Search-R1 training setup with `verl` v0.7.1, the final performance is significantly lower, even when using the same dataset, model, reward setting, and approximately the same training hyperparameters.
- In v0.8.0, the Search-R1-related implementation/examples appear to have been removed.
- I tried adapting the previous Search-R1 code to v0.8.0, but the reproduced performance still differs substantially from the earlier version.

Therefore, it seems that simply porting the old configuration and scripts to newer versions is not sufficient to reproduce the original Search-R1 results.

## Questions

I would like to ask:

1. Are there known behavioral or algorithmic changes between the earlier versions and v0.7.1/v0.8.0 that could significantly affect Search-R1 performance?

2. Besides configuration/API changes, are there changes in rollout, GRPO/PPO training, advantage computation, loss computation, sequence handling, reward processing, or the SGLang/vLLM integration that need to be taken into account when reproducing older experiments?

3. Is there an officially recommended configuration or migration guide for reproducing Search-R1 results on newer versions of `verl`?

4. Since the Search-R1 implementation has been removed in v0.8.0, what is the recommended way to reproduce Search-R1 experiments with the current codebase?

It would be very helpful to know whether similar performance regressions have been observed by others, and whether there are specific commits or configuration changes required to recover the performance of the earlier implementation.

<img width="650" height="646" alt="Image" src="https://github.com/user-attachments/assets/2d22276f-9344-437f-ba13-401330f3f391" />
The blue represents v5.0.0; the other color represents v0.7.1.

### Information

- [x] The official example scripts
- [x] My own modified scripts

### Tasks

- [x] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

the official example scripts is `verl/examples/sglang_multiturn/search_r1_like/run_qwen2.5-3b_instruct_search_multiturn.sh`

### Expected behavior

When reproducing the same Search-R1 training setup with `verl` v0.7.1, the final performance is significantly lower, even when using the same dataset, model, reward setting, and approximately the same training hyperparameters based on same official example script.
